presto-spi/src/main/java/com/facebook/presto/spi/memory/ArrayPool.java

```diff
@@ -49,16 +91,27 @@ public ByteArrayPool(int smallestSize, int largestSize, long capacity)
             contents[i] = new ArrayList();
             times[i] = new LongArrayList();
         }
+        this.allocator = requireNonNull(allocator, "allocator is null");
+    }
+
+    public ArrayType alloc(int size)
+    {
+        return alloc(size, false);
     }
 
-    public byte[] getBytes(int size)
+    public ArrayType allocInitialized(int size)
+    {
+        return alloc(size, true);
+    }
+
+    public ArrayType alloc(int size, boolean init)
```

Comment:
this method is not used outside of this class; make it private

init -> initialize